libkbfs/folder_branch_ops.go

```diff
@@ -5447,31 +5448,54 @@ func (fbo *folderBranchOps) backgroundFlusher(betweenFlushes time.Duration) {
 			// so don't bother waiting for a signal, just get right to
 			// the main attraction.
 			doSelect = false
+		} else if fbo.getCachedDirOpsCount(lState) >=
+			fbo.config.BGFlushDirOpBatchSize() {
+			doSelect = false
 		}
 
 		if doSelect {
 			// Wait until we really have a write waiting.
 			forced := false
 			select {
```

Comment:
We still need a select loop without `<-timer.C` to avoid waking up every 1 second when there are no writes.  You still want to avoid that, right?